### ISSUE TITLE
automatic restart clojure-lsp server after saving config.edn

### ISSUE THREAD

**Issue by megayu** [BetterThanTomorrow/calva#1648]
According to https://clojure-lsp.io/settings/#diagnostics-linter and https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md, we can define an config.edn file under ".clj-kondo" directory, each time i change this file i had to stop the clojure-lsp server first then start it again to make it work, it's not too convenient.

**Comment by PEZ** [comment]
Hello, thanks for the request!

It could be a bit tricky to get auto-restart to work correctly. Configs can reside in many places. And it could also be undesirable in some situations to pay the price of a full reload of clojure-lsp when you just want to save your change.

What would be pretty easy to add and keep things simple is a **Restart clojure-lsp** commands. Then you can bind it to a keyboard shortcut to easily get it done after save.

**Comment by megayu** [comment]
I think add a Restart clojure-lsp command is a good idea.

**Comment by Cyrik** [comment]
If I'm not mistaken clojure-lsp should catch changes to the config files and updated its running-config. Is this also true for the clj-kondo config @ericdallo?

**Comment by ericdallo** [comment]
yes, but there are a few things to consider, clj-kondo reads the config every time it's called, we call clj-kondo mostly on 2 places:
- when edit a single file, this should already work, changing a clj-kondo config will make next time we lint a single file consider that change.
- when starting clojure-lsp, linting whole project (and if no cache, the whole classpath), changing a clj-kondo config **wont** change these diagnostics since the project already started and linted mostly files.

Also, clojure-lsp is smart enough to know a clj-kondo config changed and invalidate the cache linting whole classpath again to avoid false-positives

**Comment by bpringe** [comment]
I've created an issue for adding a command to restart clojure-lsp, FYI: https://github.com/BetterThanTomorrow/calva/issues/1727. Someone asked if there was a way to restart it with a shortcut in Slack today.

**Comment by PEZ** [comment]
@megayu, are you aware of [Joyride](https://github.com/BetterThanTomorrow/joyride#readme)? It is an extension that @borkdude and I have created very recently, that lets you script VS Code.

Once you've installed Joyride, this keyboard shortcut restarts clojure-lsp.

```json
    {
        "key": "cmd+f4",
        "command": "joyride.runCode",
        "args": "(require '[promesa.core :as p]) (p/do (vscode/commands.executeCommand \"calva.clojureLsp.stop\") (vscode/commands.executeCommand \"calva.clojureLsp.start\"))"
    },
```

I would probably define a function for it in some `my-commands` namespace during [Joyride activate](https://github.com/BetterThanTomorrow/joyride/blob/master/doc/api.md#scripting-life-cycle) and call that function instead. This will keep the shortcut simpler. And such a namespace is a natural place to start writing code like this and get it right using the REPL. I will add an example recipe for this to Joyride, I think.

**Comment by megayu** [comment]
> 

Yes, I saw the video in YouTube before, it sounds good for me.

**Comment by bpringe** [comment]
Related issue: https://github.com/BetterThanTomorrow/calva/issues/1113
